packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -546,7 +580,9 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
         || oldDelegate.textDirection != textDirection
         || oldDelegate.floatingActionButtonMoveAnimationProgress != floatingActionButtonMoveAnimationProgress
         || oldDelegate.previousFloatingActionButtonLocation != previousFloatingActionButtonLocation
-        || oldDelegate.currentFloatingActionButtonLocation != currentFloatingActionButtonLocation;
+        || oldDelegate.currentFloatingActionButtonLocation != currentFloatingActionButtonLocation
+        || oldDelegate.extendBody != extendBody
```

Comment:
I'm glad to see this check added in. It's absence caused some head scratching while i wad developing my attempt at this feature =)

Reply:
Yes, it looks like we failed to get this right when extendBody was added. I think that's probably my fault. Sorry about the confusion!